frontend/src/components/AnnouncementsList/AnnouncementsList.js

```diff
@@ -0,0 +1,66 @@
+import { useEffect, useState } from 'react';
+import { useQuery } from 'react-query';
+import Announcement from '../Announcement/Announcement';
+import PropTypes from 'prop-types';
+
+const AnnouncementsList = ({ boardsIds, announcementsNumber, showBoardNames }) => {
+  const { data, isLoading, isError } = useQuery('boardsList', async () => {
+    return await fetch('/boards').then((response) => response.json());
+  });
+
+  const [announcements, setAnnouncemensts] = useState([]);
+  const [windowSize, setWindowSize] = useState('');
+
+  useEffect(() => {
+    const resizeHandler = () => setWindowSize(window.innerWidth);
+    window.addEventListener('resize', resizeHandler);
+
+    return () => {
+      window.removeEventListener(resizeHandler);
+    };
+  }, []);
+
+  useEffect(() => {
+    if (!!data && !!data.boards) {
+      console.log(data.boards.filter(({ boardId }) => boardsIds.includes(boardId)));
+      const announcementsToShow = data.boards
+        .filter(({ boardId }) => boardsIds.includes(boardId))
+        .reduce((acc, { announcements }) => {
+          acc.push(...announcements);
+          return acc;
+        }, [])
+        .sort((a, b) => b.date - a.date)
```

Comment:
No tutaj w sort chyba już nie ma co zmieniać nazw zmiennych, nie wiem. Rzadko się spotykam, ale jak się kiedyś dowiem to dam znać ;)

Reply:
i tak zmieniłem